### ISSUE TITLE
[feature request] name keyboard on space bar

### ISSUE THREAD

**Issue by Joppla** [Julow/Unexpected-Keyboard#664]
Maybe it is already suggested, but I couldn't find it without reading all the 133 open issues ;)

Is it possible to show the name of the active keyboard on the space-bar? As in this picture?
![Screenshot_20240611-160444_1](https://github.com/Julow/Unexpected-Keyboard/assets/25467829/0e6a7905-123d-4dae-abf9-89ab08aa12cc)

**Comment by Spike-from-NH** [comment]
If you have a custom layout file, you can do this by adding an `indication="`...`"` property to the space bar.

**Comment by Joppla** [comment]
Thanks for your answer.
It works, but doesn't look nice. Because it overwrites the s-position. It should be nicer if it overwrites the "space" symbol. The shape of the space-bar is clear enough to recognize without symbol.

**Comment by rhyton-scrag** [comment]
For me it only shows first 3 chars. For example,
 indication="Dutch querty"
shows "Dut" on the key. The space key has room for the whole string.

**Comment by Spike-from-NH** [comment]
I got by e-mail a post to this issue that the legend is truncated to 3 characters—the post presumably deleted by the author. I reported this in #613 and it is corrected in version 1.29.

**Comment by rhyton-scrag** [comment]
I apologize for neglecting to update to the latest version before posting that. Also I apologize for simply deleting the message when I realized I posted in error. I know now that I should have revised or replied saying to ignore my post.

Thank you for your contributions to Unexpected-Keyboard.

**Comment by Spike-from-NH** [comment]
Yes, @rhyton-scrag, it baffled me to try to reply to that nonexistent post, but no harm done. Do stay involved with the project!

**Comment by Spike-from-NH** [comment]
Recent observation—
•Since adding `indication` to my custom layouts with name and date, I seem to regard the legend as not part of the space bar; I'm aiming high when trying to type space! (Not a bug)
•@Julow, if Real Life is distracting too much, I wish you'd wrap up v1.30.

**Comment by Spike-from-NH** [comment]
As of v1.31, we can specify keycap legends—and they are **smaller** than the `indication`! This suits me best; I want the version ID to be unobtrusive.  On my space bar, I have `s="Spike Apr'25 P: "`.  For May'25, I'll split it up and move it to the two keys I have to the left, to avoid the problem noted above that it makes me "aim high" and miss the space bar.

![Image](https://github.com/user-attachments/assets/4be5a3a4-4ca5-44c6-82cf-798e5d73d81b)

So the south swipe also sends space. As a side-effect of the request that spaces turn to Esc if there's a cut region defined (to avoid deleting it), all these keycap legends say "Esc" too! If I remove the space from my key value, "Spike Apr'25 P" becomes the key value and it's truncated at three letters. @Julow, I would lıke to be able to have nothing after the colon to specify a swipe that does nothing but has a legend.

**Comment by Joppla** [comment]
@Spike-from-NH Thanks.
